lib/rucio/db/sqla/models.py

```diff
@@ -367,6 +367,15 @@ class DataIdentifier(BASE, ModelBase):
                    Index('DIDS_EXPIRED_AT_IDX', 'expired_at'))
 
 
+class DidMeta(BASE, ModelBase):
+    __tablename__ = 'did_meta'
+    scope = Column(String(SCOPE_LENGTH))
+    name = Column(String(NAME_LENGTH))
+    meta = Column(JSON())
+    _table_args = (PrimaryKeyConstraint('scope', 'name', name='DID_META_PK'),
+                   ForeignKeyConstraint(['scope', 'name'], ['dids.scope', 'dids.name'], ondelete='CASCADE', name='DID_META_FK'),)
```

Comment:
I am not so sure about the ondelete=CASCADE statement, as this might also have implications on some workflows we don't think about. For that reason we did not use this anywhere else. I would prefer if this is done without cascaded deletes.

Reply:
This needs to be thoroughly tested. We can't have the deletion of metadata cascade delete the original DID :-)